### DISCUSSION TITLE
Welcome to SI4684-DAB-Receiver Discussions!

### DISCUSSION THREAD

**Discussion by PE5PVB** [PE5PVB/SI4684-DAB-Receiver#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by GOUNOT** [comment]
Hello Sjef, 
After studying the DAB+ standard and texts. I learned that the first SPIbuffer [26] contained the type of text code. (UTF8 = 0xf0 UCS2 = 0x60 and ASCII Latin = 0. I, therefore, modify the part "Dab :: ascii" and use a uint8_t CodetXt in *.uno and declare externally in SI8648.H.
Sincerely. (Latin 0x60 never use)
[si4684h.txt](https://github.com/user-attachments/files/15534043/si4684h.txt)
[si4684cpp.txt](https://github.com/user-attachments/files/15534051/si4684cpp.txt)
[SI4684-DAB-Receiver.txt](https://github.com/user-attachments/files/15534038/SI4684-DAB-Receiver.txt)

Pascal

**Reply by grimpeur-DAB** [reply]
Bonjour, je voudrais construire votre radio DAB, comment faire ?, j'ai des connaissances électroniques mais pas en informatique,
Merci pour votre aide,
Cordialement, jmt
__________________________________
Hello, I would like to build your DAB radio, how to do it?, I have electronic knowledge but not in computers,
Thanks for your help,
Sincerely, jmt

**Comment by PE5PVB** [comment]
Pascal,

I used the conversion table on page 40 of 
https://www.etsi.org/deliver/etsi_ts/101700_101799/101756/01.08.01_60/ts_101756v010801p.pdf

What did you change?

Sjef

GOUNOT schreef op 2024-06-03 14:56:

**Comment by PE5PVB** [comment]
Did you make this change on my latest commit?

**Comment by PE5PVB** [comment]
I tried your modification but it mess up German![image](https://github.com/PE5PVB/SI4684-DAB-Receiver/assets/76666293/71f1b998-2639-475c-ab70-093e8966839e)

**Comment by GOUNOT** [comment]
Bonjour Sjef,Désolé pour le retard. Oui ,effectivement, les intitulés radio sont en ASCII si il n'y a pas de messages déroulant le décodage restait en utf8. J'ai corrigé l'anomalie. Dans *.ino. Au moment de l'affichage des caractéristiques.(Ligne 387) Sans incidence pour le reste.Bonne journée.Pascal[SI4684-DAB-Receiver.txt](https://github.com/user-attachments/files/15604754/SI4684-DAB-Receiver.txt)à intégrer avec gui & si4684envoyé : 5 juin 2024 à 11:51de : Sjef Verhoeven PE5PVB ***@***.***>à : PE5PVB/SI4684-DAB-Receiver ***@***.***>cc : GOUNOT ***@***.***>, Comment ***@***.***>objet : Re: [PE5PVB/SI4684-DAB-Receiver] Welcome to SI4684-DAB-Receiver Discussions! (Discussion #1) I tried your modification but it mess up German2CE8E301-C178-4D53-9413-D56316AE2B6C.jpeg (view on web)—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by PE5PVB** [comment]
I just invited you as collaborator, so you make create a pull request. This is much easier than sending the files ;=) Can you use that?

**Comment by PE5PVB** [comment]
Well, if there is only a difference between UTF8 and ASCII, it can be much more simple. I'll try to change it in a way how I handled eRT in my RDS decoder.

**Comment by marivo** [comment]
Dear Sjef, I am trying to compile version 1.10 (the latest from GitHub) but it cannot because the size of the sketch is larger than program memory (compiled with HUGE App 3MB option) The error message is related to the following output "Der Sketch verwendet 3161669 Bytes (100%) des Programmspeicherplatzes. Das Maximum sind 3145728 Bytes.". So, I am missing about 16KB for the program memory. All the other settings are as recommended (I mean for the TFT_eSPI lib). Could you give an advice how come that the original sketch does not fit into normal ESP32 4MB flash memory? Thanks!

**Comment by PE5PVB** [comment]
It should fit in 4MB ESP32 flash. Can you make a screenshot of your compiler settings?

**Comment by marivo** [comment]
That was quick, thanks! Here it goes attached
![Bildschirmfoto 2024-12-09 um 15 16 19](https://github.com/user-attachments/assets/2d56ff37-f19f-4c28-80ce-a6a5fc5dedbc)
![Bildschirmfoto 2024-12-09 um 15 21 05](https://github.com/user-attachments/assets/01bca307-7df8-4a7e-afe1-60d00393a475)

**Comment by marivo** [comment]
I compile on MacOS, btw. Further, I also had to install TimeLib because otherwise it did not compile. I know I could create my own partition of the ESP32 4MB flash, taking into account that I need about 20kB more of prog. mem but that is too much of a hassle for me... There must be something I obviously overlook when compliing because otherwise you would not be able to compile it yourself ;)

**Comment by PE5PVB** [comment]
Looks like you are using boardmanager v3.xx? Please use 2.0.17

**Comment by marivo** [comment]
Thanks Sjef, that did the trick.... Interesting to see that newer versions of ESP32 toolchain produce increasingly fatier code :) Thanks again!

**Comment by PE5PVB** [comment]
Well, I haven't changed over to v3.xx because that needs some more changes.

**Comment by marivo** [comment]
I have now successfully uploaded the sketch and I see a white screen (ILI9341 chip). I have to say that I have not yet connected any of the DAB IC itself nor I plan to connect the I2C audio chip, streaming directly I2S to a I2S-BT. Can that be that the program execution stopps if does not find /cannot read from any of DAB chip or the I2C audio chip (which might be the reason why I see the white screen)? Thanks!

**Reply by PE5PVB** [reply]
If you have a white screen, check the configurationfile of tft_espi for the right settings. See the Wiki

**Comment by marivo** [comment]
Hi Sjef, I can see that you have uploaded a few new files recently that (as it seems) concern platformIO and memory partitioning. Is that something I should consider to compile your project's sources or I can just stay with ArduinoIDE and ignore the new uploads? Thanks.

**Comment by PE5PVB** [comment]
Let's not have different discussions here. Please open a new discussion on a new topic.
